pandas/tests/tools/test_to_datetime.py

```diff
@@ -2530,3 +2530,14 @@ def test_empty_string_datetime_coerce__unit():
     # verify that no exception is raised even when errors='raise' is set
     result = to_datetime([1, ""], unit="s", errors="raise")
     tm.assert_index_equal(expected, result)
+
+
+@pytest.mark.parametrize("cache", [True, False])
+def test_to_datetime_monotonic_increasing_index(cache):
+    # GH28238
+    times = date_range(pd.Timestamp("2002"), periods=3, freq="h")
+    times = times.to_frame(index=False, name="DT").sample()
+    times.index = times.index.to_series().astype(float) / 1000
```

Comment:
`times` can be simplified to be a `Series` with non-monotonic dates:

`times = Series([pd.Timestamp('2002'), pd.Timestamp('1992'), pd.Timestamp("2022")])`

Reply:
Updated. It seems that cache is only used when there are at least 50 values. Will this matter in regards to testing?

Reply:
Yes

Reply:
Its been updated to 50 timestamps with yearly frequency.